polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -4965,9 +5009,34 @@ mlir::func::FuncOp MLIRASTConsumer::GetOrCreateMLIRFunction(
     }
   }
   mlir::OpBuilder builder(module->getContext());
-  auto funcType = builder.getFunctionType(types, rettypes);
-  mlir::func::FuncOp function = mlir::func::FuncOp(mlir::func::FuncOp::create(
-      getMLIRLocation(FD->getLocation()), name, funcType));
+
+  const auto createFunctionDecl = [&]() -> FunctionOpInterface {
+    const auto insert = [&](auto Function, auto Module, auto &Map) {
+      Module.push_back(Function);
+      Map[name] = Function;
+      return Function;
+    };
+    auto funcType = builder.getFunctionType(types, rettypes);
```

Comment:
auto -> mlir::FunctionType ?

Reply:
I think in this case this isn't needed, as it is quite straightforward that `mlir::OpBuilder::getFunctionType` should return a `mlir::FunctionType`